packages/@glimmer/application/src/application.ts

```diff
@@ -129,22 +148,122 @@ export type Notifier = [() => void, (err: Error) => void];
 const DEFAULT_DOCUMENT = typeof document === 'object' ? document : null;
 
 /**
- * The central control point for starting and running Glimmer components.
+ * A Base Application Class that can be shared across different glimmer application implementations.
+ * It sets up a dependency injection and sets up the necessary arguments for any glimmer application.
+ * For example this base class can be reused across a server side glimmer application and a client side glimmer application.
  *
- * @public
+ * @internal
  */
-export default class Application implements Owner {
+export class BaseApplication implements Owner {
```

Comment:
I think I might be a little bit confused. Don't we need a "base" object that is responsible for setting up the initial registry then effectively becoming a factory that produces clean Application instances that reference the registry? I would assume that we don't want to do all registry setup per request in the SSR case.

Reply:
The idea behind the BaseApplication was exactly this. The way it is setup right now the SSRApplication instance can be reused across requests and it does not setup on a new registry per request. Specifically I mean the renderToStream and renderToString method can be safely called on the same application instance across different requests.